java/dataset/src/test/java/org/apache/arrow/dataset/ParquetWriteSupport.java

```diff
@@ -42,13 +43,15 @@
   private final Schema avroSchema;
   private final List<GenericRecord> writtenRecords = new ArrayList<>();
   private final GenericRecordListBuilder recordListBuilder = new GenericRecordListBuilder();
+  private final Random random = new Random();
 
 
   public ParquetWriteSupport(String schemaName, File outputFolder) throws Exception {
     avroSchema = readSchemaFromFile(schemaName);
-    path = outputFolder.getPath() + File.separator + "generated.parquet";
+    path = outputFolder.getPath() + File.separator + "generated-" + random.nextLong() + ".parquet";
```

Comment:
I think that this change wants to get a unique name for a short period.

How about using `System.currentTimeMillis()`?

Reply:
Doesn't milli cause collisions? `ParquetWriteSupport` is instantiated rapidly during ut execution.

By the way any special reason to use time rather than random? The files can be dropped after finishing ut. Users should not be aware of the files.